### ISSUE TITLE
Helm Chart and Docs do not mention `aws.enableInterruptionHandling` setting

### ISSUE THREAD

**Issue by snorlaX-sleeps** [aws/karpenter#2897]
### Is an existing page relevant?

https://karpenter.sh/v0.19.0/tasks/globalsettings/#configmap

### What karpenter features are relevant?

Interruption Queue Handling by Karpenter v0.19+

### How should the docs be improved?

[This PR](https://github.com/aws/karpenter/pull/2855/files) removed the `enableInterruptionHandling` setting from the documentation, instead replacing it with the `interruptionQueueName` setting, which was missing.

However, this variable still appears in the `karpenter-global-settings` configmap, even when it is not set by the user. (defaults to false)

Is this variable required? If so it should be documented to enable users to use the SQS Interruption handling (I believe it is, as when it was set to true logs would appear around the functionality)

If it's not required to enable the feature, can it be removed from the generated configmap to avoid confusion? 

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by FernandoMiguel** [comment]
I'm also very curious and confused by what is happening with NSTH ... the entire handling of SQS has changed and I don't see any explanation for it

**Comment by snorlaX-sleeps** [comment]
@FernandoMiguel you can still use the AWS NTH as per previous versions, but they have added new in-built functionality as documented [here](https://karpenter.sh/v0.19.0/tasks/deprovisioning/#interruption) (does not yet support rebalance events).

The [eksctl cloudformation template](https://github.com/aws/karpenter/blob/main/website/content/en/v0.19.1/getting-started/getting-started-with-eksctl/cloudformation.yaml) has examples for updating the IAM permissions and deploying the SQS Queue, which are the same as the old way just attached to Karpenter, rather than the separate deployment

**Comment by FernandoMiguel** [comment]
@snorlaX-sleeps i kept tracking the initial work, and even wrote the terraform code to match the cloudformation IAM changes 
https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/1145

it is just not clear to me at all what happened after the initial work.
lot's of permissions were removed , so it cant create those resources, instead it seems limited to read/write into existing queues, which kinda defeats the initial purpose.

**Comment by bwagner5** [comment]
We pivoted from our initial implementation where Karpenter would create the infrastructure because of some feedback around compliance of the infrastructure and that users would rather create the infra themselves.  That is why some permissions were removed at the last minute and Karpenter no longer creates the SQS queue and EventBridge rules. Instead, Karpenter accepts the `interruptionQueueName` in the configmap which should be the SQS queue that you have setup. If the queue name is passed, then Karpenter implicitly starts to consume messages from that queue.  If the `enableInterruptionHandling` parameter is still being populated in the configmap, then we need to remove that, since it shouldn't be used anywhere.
